lantern.nsi

```diff
@@ -0,0 +1,86 @@
+Name "Lantern"
+
+# Installs Lantern and launchs it
+# See http://nsis.sourceforge.net/Run_an_application_shortcut_after_an_install
+
+AutoCloseWindow true
+
+!addplugindir nsis_plugins
+!include "nsis_includes/nsProcess.nsh"
+
+# Use the modern ui
+!include MUI.nsh
+!define MUI_ICON lantern.ico
+
+;Languages
+!insertmacro MUI_LANGUAGE "Farsi"
+!insertmacro MUI_LANGUAGE "English"
+!insertmacro MUI_LANGUAGE "Czech"
+!insertmacro MUI_LANGUAGE "Dutch"
+!insertmacro MUI_LANGUAGE "French"
+!insertmacro MUI_LANGUAGE "German"
+!insertmacro MUI_LANGUAGE "Korean"
+!insertmacro MUI_LANGUAGE "Russian"
+!insertmacro MUI_LANGUAGE "Spanish"
+!insertmacro MUI_LANGUAGE "Swedish"
+!insertmacro MUI_LANGUAGE "TradChinese"
+!insertmacro MUI_LANGUAGE "SimpChinese"
+!insertmacro MUI_LANGUAGE "Slovak"
+
+# define name of installer
+OutFile "lantern-installer-unsigned.exe"
+ 
+# define installation directory
+InstallDir $PROGRAMFILES\Lantern
+ 
+# For removing Start Menu shortcut in Windows 7
+RequestExecutionLevel admin
+    
+# start default section
+Section
+    # Stop existing Lantern if necessary
+    ${nsProcess::KillProcess} "lantern.exe" $R0
+    # Sleep for 1 second to process a chance to die and file to become writable
+    Sleep 1000
+
+    # Remove anything that may currently be installed
+    RMDir /r "$SMPROGRAMS\Lantern"
+    RMDir /r "$INSTDIR"
+    
+    # set the installation directory as the destination for the following actions
+    SetOutPath $INSTDIR
+
+    File lantern.exe
+    File lantern.ico
+ 
+    # Store installation folder
+    WriteRegStr HKCU "Software\Lantern" "" $INSTDIR
+
+    WriteUninstaller "$INSTDIR\uninstall.exe"
+
+    # Support uninstalling via Add/Remove programs
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Lantern" \
+                     "DisplayName" "Lantern"
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Lantern" \
+                     "UninstallString" "$\"$INSTDIR\uninstall.exe$\""
```

Comment:
That's actually what I did at first, but having the installer and the uninstaller use the same icon didn't look quite right.  If we design an uninstall icon of our own, then this would be cool, but for now I figured the default uninstaller icon is good enough.